tools/functional-tester/etcd-agent/client/client.go

```diff
@@ -48,7 +52,16 @@ type agent struct {
 }
 
 func NewAgent(endpoint string) (Agent, error) {
-	c, err := rpc.DialHTTP("tcp", endpoint)
+	url, err := url.Parse(endpoint)
+	if err != nil {
+		return nil, err
+	}
+	_, port, err := net.SplitHostPort(url.Host)
+	if err != nil {
+		return nil, err
+	}
+
+	c, err := rpc.DialHTTP("tcp", ":"+port)
```

Comment:
when you pass multiple endpoints as a string slice e.g. `http://localhost:1234`, you need to pass `:1234` to `rpc.DialHTTP`. Otherwise, it panics with `too many colons`.